pkg/registry/apis/dashboard/register.go

```diff
@@ -170,144 +173,112 @@ func (b *DashboardsAPIBuilder) Validate(ctx context.Context, a admission.Attribu
-		return &dashboardv0alpha1.DashboardWithAccessInfo{}
-	})
-	if err != nil {
+	if err := b.storageForVersion(apiGroupInfo, opts, largeObjects,
+		dashboardv0alpha1.DashboardResourceInfo,
+		dashboardv0alpha1.LibraryPanelResourceInfo,
+		func() runtime.Object {
+			return &dashboardv0alpha1.DashboardWithAccessInfo{}
+		}); err != nil {
 		return err
 	}
-	apiGroupInfo.VersionedResourcesStorageMap[dashboardv0alpha1.VERSION] = storage
 
 	// v1alpha1
-	storage, err = b.storageForVersion(opts, legacyStore, largeObjects, func() runtime.Object {
-		return &dashboardv1alpha1.DashboardWithAccessInfo{}
-	})
-	if err != nil {
+	if err := b.storageForVersion(apiGroupInfo, opts, largeObjects,
+		dashboardv1alpha1.DashboardResourceInfo,
+		dashboardv1alpha1.LibraryPanelResourceInfo,
+		func() runtime.Object {
+			return &dashboardv1alpha1.DashboardWithAccessInfo{}
+		}); err != nil {
 		return err
 	}
-	apiGroupInfo.VersionedResourcesStorageMap[dashboardv1alpha1.VERSION] = storage
 
 	// v2alpha1
-	storage, err = b.storageForVersion(opts, legacyStore, largeObjects, func() runtime.Object {
-		return &dashboardv2alpha1.DashboardWithAccessInfo{}
-	})
-	if err != nil {
+	if err := b.storageForVersion(apiGroupInfo, opts, largeObjects,
+		dashboardv2alpha1.DashboardResourceInfo,
+		dashboardv2alpha1.LibraryPanelResourceInfo,
+		func() runtime.Object {
+			return &dashboardv2alpha1.DashboardWithAccessInfo{}
+		}); err != nil {
 		return err
 	}
-	apiGroupInfo.VersionedResourcesStorageMap[dashboardv2alpha1.VERSION] = storage
 
 	return nil
 }
 
 func (b *DashboardsAPIBuilder) storageForVersion(
+	apiGroupInfo *genericapiserver.APIGroupInfo,
 	opts builder.APIGroupOptions,
-	legacyStore grafanarest.Storage,
 	largeObjects apistore.LargeObjectSupport,
+	dashboards utils.ResourceInfo,
+	libraryPanels utils.ResourceInfo,
 	newDTOFunc func() runtime.Object,
-) (map[string]rest.Storage, error) {
-	var (
-		err                      error
-		scheme                   = opts.Scheme
-		dualWriteBuilder         = opts.DualWriteBuilder
-		internalDashResourceInfo = dashboardinternal.DashboardResourceInfo
-		libraryPanelResourceInfo = dashboardinternal.LibraryPanelResourceInfo
-	)
-
+) error {
+	// Register the versioned storage
 	storage := map[string]rest.Storage{}
-	storage[internalDashResourceInfo.StoragePath()] = legacyStore
+	apiGroupInfo.VersionedResourcesStorageMap[dashboards.GroupVersion().Version] = storage
 
-	// Dual writes if a RESTOptionsGetter is provided
-	if dualWriteBuilder != nil {
-		store, err := grafanaregistry.NewRegistryStore(scheme, internalDashResourceInfo, opts.OptsGetter)
-		if err != nil {
-			return nil, err
-		}
-		storage[internalDashResourceInfo.StoragePath()], err = dualWriteBuilder(internalDashResourceInfo.GroupResource(), legacyStore, store)
-		if err != nil {
-			return nil, err
-		}
+	if true {
+		dashboards = dashboardinternal.DashboardResourceInfo // use the same version
```

Comment:
^^ this flag lets me switch between having all storages registered with internal vs explicit versions.

When internal, every get request will convert twice:
<img width="788" alt="image" src="https://github.com/user-attachments/assets/99279bdb-b49d-46d2-afb5-8be1e901188b" />